smac_planner/src/a_star.cpp

```diff
@@ -282,11 +286,12 @@ bool AStarAlgorithm<NodeT>::createPath(
   float g_cost;
   NodeVector neighbors;
   int approach_iterations = 0;
+  int analytic_iterations = 0;
   typename NodeVector::iterator neighbor_iterator;
 
   // Given an index, return a node ptr reference if its collision-free and valid
   const unsigned int max_index = getSizeX() * getSizeY() * getSizeDim3();
-  std::function<bool(const unsigned int &, NodeT * &)> neighbor_getter =
+  ValidityChecker neighbor_getter =
```

Comment:
This isn't a validity checker, this is only getting the neighbors - see the lambda. Its just returning a pointer to the node in the graph it was looking for.

Reply:
I have renamed to better reflect its function.